libarchive fails to build on Tumbleweed with gcc 8.1.1

Hi,

I just tried to build this project, I will confess I did manually bump the submodules (avogadroapp, avogadrolib, avogadrodata, molequeue) to their latest master commit, but given that this build failure is in a thirdparty library I don't think that is relevant. The exact error is:

```bash
-- Build files have been written to: /home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src
[ 57%] Performing build step for 'libarchive'
Scanning dependencies of target archive_static
[  0%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_acl.c.o
[  0%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_check_magic.c.o
[  1%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_cmdline.c.o
[  1%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_cryptor.c.o
[  1%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_digest.c.o
[  1%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_entry.c.o
[  1%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_entry_copy_stat.c.o
[  1%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_entry_link_resolver.c.o
[  1%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_entry_sparse.c.o
[  2%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_entry_stat.c.o
[  2%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_entry_strmode.c.o
[  2%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_entry_xattr.c.o
[  2%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_getdate.c.o
[  2%] Building C object libarchive/CMakeFiles/archive_static.dir/archive_hmac.c.o
In file included from /home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_hmac_private.h:73,
                 from /home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_hmac.c:32:
/home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_openssl_hmac_private.h:34:25: error: static declaration of ‘HMAC_CTX_new’ follows non-static declaration
 static inline HMAC_CTX *HMAC_CTX_new(void)
                         ^~~~~~~~~~~~
In file included from /home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_openssl_hmac_private.h:28,
                 from /home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_hmac_private.h:73,
                 from /home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_hmac.c:32:
/usr/include/openssl/hmac.h:86:11: note: previous declaration of ‘HMAC_CTX_new’ was here
 HMAC_CTX *HMAC_CTX_new(void);
           ^~~~~~~~~~~~
In file included from /home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_hmac_private.h:73,
                 from /home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_hmac.c:32:
/home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_openssl_hmac_private.h:40:20: error: static declaration of ‘HMAC_CTX_free’ follows non-static declaration
 static inline void HMAC_CTX_free(HMAC_CTX *ctx)
                    ^~~~~~~~~~~~~
In file included from /home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_openssl_hmac_private.h:28,
                 from /home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_hmac_private.h:73,
                 from /home/fusion809/GitHub/others/openchemistry-build/thirdparty/libarchive-src/libarchive/archive_hmac.c:32:
/usr/include/openssl/hmac.h:87:6: note: previous declaration of ‘HMAC_CTX_free’ was here
 void HMAC_CTX_free(HMAC_CTX *ctx);
      ^~~~~~~~~~~~~
gmake[5]: *** [libarchive/CMakeFiles/archive_static.dir/build.make:232: libarchive/CMakeFiles/archive_static.dir/archive_hmac.c.o] Error 1
gmake[4]: *** [CMakeFiles/Makefile2:1021: libarchive/CMakeFiles/archive_static.dir/all] Error 2
gmake[3]: *** [Makefile:141: all] Error 2
gmake[2]: *** [thirdparty/CMakeFiles/libarchive.dir/build.make:114: thirdparty/libarchive-prefix/src/libarchive-stamp/libarchive-build] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:288: thirdparty/CMakeFiles/libarchive.dir/all] Error 2
gmake: *** [Makefile:84: all] Error 2
```

openssl version is 1.1.0h-fips. Kernel version, if it somehow relevant is 4.17.3-1-default. cmake version is 3.11.4. My openSUSE Tumbleweed version is 20180707. If there's any other bits of information you need feel free to ask and I'll be more than happy to provide it. 

Thanks for your time,
Brenton

--- a/cmake/projects.cmake
+++ b/cmake/projects.cmake
@@ -69,9 +69,9 @@ set(spglib_md5 "1933e2252a0e708951ee476eb93f6495")
 
 # libarchive
 list(APPEND projects libarchive)
-set(libarchive_version "3.3.2")
+set(libarchive_version "3.5.1")
 set(libarchive_url "https://github.com/libarchive/libarchive/archive/v${libarchive_version}.tar.gz")
-set(libarchive_md5 "a3acebe237f89d7f31c5bb1da5e843c7")
+set(libarchive_md5 "742a47936955f280b9dc712ac544f8ec")
 
 # msgpackc
 list(APPEND projects msgpackc)

